tidb-lightning/tidb-lightning-faq.md

```diff
@@ -187,10 +183,152 @@ See also [How to properly restart TiDB Lightning?](#how-to-properly-restart-tidb
+3. If there is a quota on CPU (for example, limited by K8s settings), Lightning may not be able to read this out. In this case, `region-concurrency` must also be **manually** reduced.
+
+**Cause 2**: The table is too complex.
+
+Every additional index will introduce a new KV pair for each row. If there are N indices, the actual size to be imported would be approximately (N+1) times the size of the Dumpling output. If the indices are negligible, you may first remove them from the schema, and add them back via `CREATE INDEX` after import is complete.
+
+**Cause 3**: Each file is too large.
+
+TiDB Lightning works the best when the data source are broken down into multiple files of size around 256 MB, so that the data can be processed in parallel. Lightning seems unresponsive if each file is too large.
+
+If the data source is CSV, and all CSV files have no fields containing literal new line characters (U+000A and U+000D), you can turn on "strict format" to let Lightning automatically split the large files.
+
+```toml
+[mydumper]
+strict-format = true
+```
+
+**Cause 4**: TiDB Lightning is too old.
+
+Try the latest version! Maybe there is new speed improvement.
+
+## `checksum failed: checksum mismatched remote vs local`
+
+**Cause**: The checksum of a table in the local data source and the remote imported database differ. This error has several deeper reasons:
+
+1. The table might already have data before. These old data can affect the final checksum.
+
+2. If the remote checksum is 0, which means nothing is imported, it is possible that the cluster is too hot and fails to take in any data.
+
+3. If the data is mechanically generated, ensure it respects the constrains of the table:
+
+    * `AUTO_INCREMENT` columns need to be positive, and do not contain the value "0".
+    * The UNIQUE and PRIMARY KEYs must have no duplicated entries.
+
+4. If TiDB Lightning has failed before and was not properly restarted, a checksum mismatch may happen due to data being out-of-sync.
+
+**Solutions**:
+
+1. Delete the corrupted data with via `tidb-lightning-ctl`, and restart Lightning to import the affected tables again.
+
+    ```sh
+    tidb-lightning-ctl --config conf/tidb-lightning.toml --checkpoint-error-destroy=all
+    ```
+
+2. Consider using an external database to store the checkpoints (change `[checkpoint] dsn`) to reduce the target database's load.
+
+3. If TiDB Lightning was improperly restarted, see also the "[How to properly restart TiDB Lightning](/tidb-lightning/tidb-lightning-faq.md#how-to-properly-restart-tidb-lightning)" section in the FAQ.
+
+## `Checkpoint for … has invalid status:` (error code)
+
+**Cause**: [Checkpoint](/tidb-lightning/tidb-lightning-checkpoints.md) is enabled, and Lightning or Importer has previously abnormally exited. To prevent accidental data corruption, Lightning will not start until the error is addressed.
```

Comment:
```suggestion
**Cause**: [Checkpoint](/tidb-lightning/tidb-lightning-checkpoints.md) is enabled, and TiDB Lightning or TiKV Importer has previously abnormally exited. To prevent accidental data corruption, Lightning will not start until the error is addressed.
```